src/devices/Qmc5883l/Oversampling.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+
+namespace Iot.Device.Qmc5883l
+{
+    /// <summary>
+    /// Over sample Rate (OSR) registers are used to control bandwidth of an internal digital filter.
+    /// Larger OSR valueleads to smaller filter bandwidth, less in-band noise and higher power consumption.It could be used to reach a
+    /// good balance between noise and power. Four over sample ratio can be selected, 64, 128, 256 or 512.
+    /// </summary>
+    [Flags]
+    public enum Oversampling : byte
+    {
+        /// <summary>
+        /// Over sample rate of 512
+        /// </summary>
+        OS512 = 0x00,
```

Comment:
We tend not to use abbreviation, so for those one, `Rate512` may work as the enum name already contains `Oversampling`, so it makes it easy to understand `Oversampling.Rate512`